### DISCUSSION TITLE
Mojo vs Quasar

### DISCUSSION THREAD

**Discussion by mvharlan** [mojolicious/mojo#1943] [General]
Hi,
I like Mojolicious because it's SBOM contains only ONE entry! I also like quasar/vue because it's neat and easy to use. But it's SBOM has some 1000++ entries. 
So I want to do sensitive stuff within perl/mojo and use npm/quasar for the UI stuff. The npm server should forward all REST calls to mojo and handle js/css/html/media/... itself.
My questions are: 
- Is this a good idea? 
- Is there a better solution?
- Should mojo serve all the npm/webpack stuff or is nodejs good enough for that?
- Are there any examples for my architecture?
Best regards,
Magnus

**Comment by rabbiveesh** [comment]
I use quasar for the frontend of all my projects (personal + work) with
mojo in the back. Highly recommended, both are progressive (in a non
political way, lol) and thus quite pleasant to build + learn with.
I have mojo serve all the static files, you just need to set them up in the
right directories.
Though I don't quite see exactly what you're asking. What problems are you
forseeing?

**Reply by mvharlan** [reply]
After starting my first vue project within a active project I asked git to commit a change in my active project. It complained about some 20000 new files in the source tree. That really scared me like hell, because nobody can control all this code and js code is not known for it's quality. Coming from a high security background this is  really s nightmare. That's the reason I would like to separate huge npm dependency trees from the really sensitive stuff using crypto, dbs, auth, etc...

**Reply by fskale** [reply]
Do you have a useful link how to integrate a quasar app into mojo ?
Since it's nodjs based, a server is forked to serve the logic.
Would be very helpful, for starters…
Best regards
Franz

**Comment by rabbiveesh** [comment]
Well, you don't have to build your frontend on the same box where sensitive
stuff is happening. Nor does it have to be in the same repo as your backend.
It compiles ultimately to static files that only run on the client (not
including SSR, but that could also be served on a different box).

**Reply by mvharlan** [reply]
That's exactly my plan.

**Comment by rabbiveesh** [comment]
Also, you should add node_modules to your gitignore

**Reply by mvharlan** [reply]
I learned that quite early  (-:

**Comment by rabbiveesh** [comment]
Quasar isn't node based (unless you're using SSR).
You use the quasar build command to minify + whatnot your files into a
bunch of static files, which you can serve as static files.

**Comment by kraih** [comment]
Btw. [mojo.js](https://github.com/mojolicious/mojo.js) now also exists. 😉
